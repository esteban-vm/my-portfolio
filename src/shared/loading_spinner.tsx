import { PacmanLoader } from 'react-spinners'

type LoadingSpinnerProps = Parameters<typeof PacmanLoader>[0]

export default function LoadingSpinner(props: LoadingSpinnerProps) {
  return <PacmanLoader {...props} aria-label='Loading Spinner' color='#00e0c0' speedMultiplier={2} />
}
