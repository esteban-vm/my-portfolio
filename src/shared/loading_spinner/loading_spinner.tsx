import { PacmanLoader } from 'react-spinners'

type LoadingSpinnerProps = Parameters<typeof PacmanLoader>[0]

export default function LoadingSpinner(props: LoadingSpinnerProps) {
  return <PacmanLoader aria-label='Loading Spinner' color='#ff15bc' speedMultiplier={2} {...props} />
}
