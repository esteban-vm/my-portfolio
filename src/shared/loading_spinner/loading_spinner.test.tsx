import { render, screen } from '@/tests'
import LoadingSpinner from './loading_spinner'

describe(`${LoadingSpinner.name}:`, () => {
  it('should render correctly', () => {
    const { asFragment } = render(<LoadingSpinner />)
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByLabelText('Loading Spinner')).toBeInTheDocument()
  })
})
