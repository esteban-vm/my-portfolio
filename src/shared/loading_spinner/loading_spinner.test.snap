// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner: > should render correctly 1`] = `
<DocumentFragment>
  <span
    aria-label="Loading Spinner"
    style="display: inherit; position: relative; font-size: 0px; height: 50px; width: 50px;"
  >
    <span
      style="width: 0px; height: 0px; border-right: 25px solid transparent; border-top: 25px solid transparent; border-left: 25px solid #ff15bc; border-bottom: 25px solid #ff15bc; border-radius: 25px; position: absolute; animation: react-spinners-PacmanLoader-pacman-1 0.4s infinite ease-in-out; animation-fill-mode: both;"
    />
    <span
      style="width: 0px; height: 0px; border-right: 25px solid transparent; border-top: 25px solid #ff15bc; border-left: 25px solid #ff15bc; border-bottom: 25px solid transparent; border-radius: 25px; position: absolute; animation: react-spinners-PacmanLoader-pacman-2 0.4s infinite ease-in-out; animation-fill-mode: both;"
    />
    <span
      style="width: 8.333333333333334px; height: 8.333333333333334px; background-color: rgb(255, 21, 188); margin: 2px; border-radius: 100%; transform: translate(0, -6.25px); position: absolute; top: 25px; left: 100px; animation: react-spinners-PacmanLoader-ball 0.5s 0.5s infinite linear; animation-fill-mode: both;"
    />
    <span
      style="width: 8.333333333333334px; height: 8.333333333333334px; background-color: rgb(255, 21, 188); margin: 2px; border-radius: 100%; transform: translate(0, -6.25px); position: absolute; top: 25px; left: 100px; animation: react-spinners-PacmanLoader-ball 0.5s 0.75s infinite linear; animation-fill-mode: both;"
    />
    <span
      style="width: 8.333333333333334px; height: 8.333333333333334px; background-color: rgb(255, 21, 188); margin: 2px; border-radius: 100%; transform: translate(0, -6.25px); position: absolute; top: 25px; left: 100px; animation: react-spinners-PacmanLoader-ball 0.5s 1s infinite linear; animation-fill-mode: both;"
    />
    <span
      style="width: 8.333333333333334px; height: 8.333333333333334px; background-color: rgb(255, 21, 188); margin: 2px; border-radius: 100%; transform: translate(0, -6.25px); position: absolute; top: 25px; left: 100px; animation: react-spinners-PacmanLoader-ball 0.5s 1.25s infinite linear; animation-fill-mode: both;"
    />
  </span>
</DocumentFragment>
`;
